### ISSUE TITLE
Stream Output Support

### ISSUE THREAD

**Issue by doitsujin** [doitsujin/dxvk#695]
With the [Vulkan 1.1.88](https://github.com/KhronosGroup/Vulkan-Docs/commit/0a7a04f32bd473bc7428efdbbbe132f33afad68c) spec update including the new `VK_EXT_transform_feedback` extension, DXVK can finally support Stream Output, which has been a big issue for a number of games (see #135).

### Prerequisites
**All** of the following are required in order for Stream Output to work. If any of these requirements are not met, you will see log messages indicating that Stream Output support is not present.
**Please make sure that your setup is working correctly before reporting bugs.**

**Note**: Proton 3.16 already includes the required patches for Wine and DXVK, so you will only need to update your Vulkan driver in order to use it with Proton.

#### DXVK
DXVK enables Stream Output support starting with release [v0.90](https://github.com/doitsujin/dxvk/releases/tag/v0.90).

#### Drivers
You will need a driver which implements the `VK_EXT_transform_feedback` extension:
- **AMD**: RADV supports the extension in Mesa 18.3.
- **Intel**: A patch series for the ANV driver is available [here](https://patchwork.freedesktop.org/series/50969/).
- **Nvidia**: Mainline driver version 418.22 supports the extension.

#### Wine
Wine 3.18 and earlier versions need to be patched in order to expose the extension to DXVK:
https://source.winehq.org/patches/data/152604

**Note**: Current development builds no longer require this patch. On Arch Linux, you may use [wine-tkg-git](https://github.com/Tk-Glitch/PKGBUILDS/tree/master/wine-tkg-git).

### Screenshot
![bildschirmfoto-356](https://user-images.githubusercontent.com/25567304/46905976-dfb38f00-cefb-11e8-90f9-e7304ac73f85.png)
**Witcher 3:** Geralt fighting the previously problematic Sirerens, with Hairworks enabled.

**Comment by pedrofleck** [comment]
Wow! I think the time to ditch the Windows partition has come. But I'll wait for the final version of Wine and Nvidia drivers with support, anyone know the ETA for those?

**Comment by doitsujin** [comment]
Wine should hopefully come with updated winevulkan in the next release (3.19, probably in two weeks). Not sure when official driver releases will land, might still take a while.

**Comment by Jrugia** [comment]
@doitsujin Brilliant work, THANK YOU!!! and to everyone involved into making this happen.

If anyone needs, I have uploaded a pre-patched version of Wine 3.18 with the Stream Output.

Wine 3.18 Source: https://drive.google.com/file/d/1SPFB71dZCejKVLxwIrkGaBQiyStPd9eb

Pre-compiled Build 3.18 with StreamOutput (Built using Debian 9.5): https://drive.google.com/file/d/1Eoaydn1XbwK0UWpn8swZ7DKcdrF1Ybxy

I haven't tried Witcher 3 yet however here's a screenshot of Vikings: Wolves of Midgard with DXVK, the main character and all enemies didn't render at all previously.

![vikings](https://user-images.githubusercontent.com/13750621/46907444-b6eac400-cf12-11e8-8fe4-c8ab3abf6b0b.jpg)

**Comment by j3kyl** [comment]
Oooooooooooohh, boy!

**Comment by asoltesz** [comment]
Huge thanks for this !!!

Can't wait to get home and try it on the Witcher.

**Comment by C0rn3j** [comment]
>Nvidia: Use the Vulkan beta version 396.54.09.

How about those on 410.XX already? Either downgrade to the 396 beta or wait for one to be released for 410.xx?

**Comment by doitsujin** [comment]
@MinIsMin The option is still used for users who don't have a driver/wine version with Transform Feedback support.

@C0rn3j The 396 beta driver is currently the only one which implements the extension. But it also has performance improvements which have not yet landed in the main branch, so it might be worth checking out either way (unless you have a Turing GPU).

**Comment by shmerl** [comment]
Rotfiends working:

![tw3_rotfiends](https://user-images.githubusercontent.com/310927/46912465-304eda80-cf44-11e8-814a-9831a99e4c52.png)

**Comment by thirdeyefunction** [comment]
Giana Sisters : Twisted Dreams crashes with RADV driver with transform feedback enabled (required by the game). Previous versions of DXVK (0.80 and 0.81 tested) work, but with broken output. The feature is working correctly on my system since I can successfully run Witcher 3 with correctly rendered rotfiends and hairworks.

Here's the error:

> SPIR-V parsing FAILED:                                                                                                                                                                                                                                        
    In file ../mesa-9999/src/compiler/spirv/vtn_variables.c:1952
    base_val->value_type == vtn_value_type_pointer
    1244 bytes into the SPIR-V binary

**Comment by doitsujin** [comment]
Might be a bug in my shader compiler, I'll have to take a look at it.

**Comment by dhewg** [comment]
Very nice to see this done!
![w3siren](https://user-images.githubusercontent.com/605548/46913768-bf371380-cf93-11e8-8594-2234e768f9b0.png)

**Comment by dimgel** [comment]
Hurrah! Hurrah! HURRAH! News of the year. :) THANK YOU!

**Comment by liberodark** [comment]
> @doitsujin Brilliant work, THANK YOU!!! and to everyone involved into making this happen.
> 
> If anyone needs, I have uploaded a pre-patched version of Wine 3.18 with the Stream Output.
> 
> Wine 3.18 Source: https://drive.google.com/file/d/1SPFB71dZCejKVLxwIrkGaBQiyStPd9eb
> 
> Pre-compiled Build with StreamOutput (using Debian 9.5): https://drive.google.com/file/d/1Eoaydn1XbwK0UWpn8swZ7DKcdrF1Ybxy
> 
> I haven't tried Witcher 3 yet however here's 2 screenshots of Vikings: Wolves of Midgard with DXVK, the main character and all enemies didn't render at all previously.
> 
> ![vikings](https://user-images.githubusercontent.com/13750621/46907444-b6eac400-cf12-11e8-8fe4-c8ab3abf6b0b.jpg)
> 
> ![troll](https://user-images.githubusercontent.com/13750621/46907564-c1a65880-cf14-11e8-9ad3-f1c320e6b69d.jpg)

Sorry but your game is on linux for what try this game with dxvk ?

**Comment by ahmed-elsayed2017** [comment]
What Mesa version should I use with AMD?

**Comment by doitsujin** [comment]
@liberodark Because it's a test case for Stream Output? Please don't start any off-topic Linux Native vs Wine discussions.

**Comment by ahmed-elsayed2017** [comment]
I am sorry, but how to apply the patch to wine manually?

**Comment by ahmed-elsayed2017** [comment]
I feel confused everytime someone talk about technical stuff, so I need to ask some questions if you excuse me:

1- How to apply a patch to Wine manually?
2- What Mesa version that support VK_EXT_transform_feedback extension?
3- Does esync-staging-ppa-3.18 have this patch?

**Comment by Tk-Glitch** [comment]
1- Like any other patch. Google has all the answers for that.
2- Devel, with some added patches that aren't yet merged in master.
3- Yes.

**Comment by Jrugia** [comment]
@liberodark  
> Sorry but your game is on linux for what try this game with dxvk ?

Just to keep to keep it short - the Linux version is not available on GOG.

**Comment by tbe** [comment]
THIS is just awesome! Thank you @doitsujin and everyone who helped to make this possible!

**Comment by asoltesz** [comment]
Stream Output seems to be working properly with the GOG version of The Witcher 3 on an AMD RX580 8GB, Kubuntu 18.04 x64 (KDE Plasma 5.13). 

Manually built Wine prefix with:
- DXVK 0.90 release
- RADV via Mesa 8.3 (current master branch + the referenced patch series applied manually (only patches 02 - 15, the first is already applied on master) + local mesa build+install). 
- Jrugia's pre-built Win 3.18 with the StreamOutput patches (https://drive.google.com/file/d/1Eoaydn1XbwK0UWpn8swZ7DKcdrF1Ybxy).

I tested on Skellige (lot of sirens and ekhidnas).

The game also seems to be faster by ~5-10% but that may be because I have updated from an older DXVK version (0.42).

**Comment by ahmed-elsayed2017** [comment]
I installed mesa-git 18.3 from AUR (it took 9 hours to compile), and it updated mesa+vulkan-radeon+vulkan-intel, then I updated vulkan-icd-loader to the latest version I found 1.1.85+2969+5abee6173-1, then I installed esync-staging-ppa-3.18 that contains the latest patches, but DXVK_HUD shows Vulkan version 1.0.68! What did I do wrong?!

**Comment by doitsujin** [comment]
Don't know, please post your full DXVK logs.

Also, there's no way that compiling mesa-git takes 9 hours, except maybe if you did it on a first-gen Raspberry Pi. Any modern desktop CPU can do it in a few minutes, so I'm not sure what exactly you compiled there.

Also, please **ignore** the Vulkan version. It means absolutely nothing, and you don't need to update the ICD loader.

**Comment by ahmed-elsayed2017** [comment]
[witcher3_d3d11.log](https://github.com/doitsujin/dxvk/files/2478410/witcher3_d3d11.log)
[witcher3_dxgi.log](https://github.com/doitsujin/dxvk/files/2478411/witcher3_dxgi.log)

I went to the sea looking for Sirens, but it is still invisible for me.

**Comment by HanFox** [comment]
You haven't said your graphics stack, but AUR mesa-git doesn't have any of the patches added. They haven't been pulled to master yet, you have to add a prepare() section to the PKGBUILD that applies the relevant patches.

**Comment by ahmed-elsayed2017** [comment]
I don't know how to do that. I will wait for the next update from Mesa that contains this patch.

**Comment by ahmed-elsayed2017** [comment]
@doitsujin  I asked about the time it took me to compile Mesa-git on Manjaro forum, and an admin there told me: "Sounds about right.", so I guess it is ok.

**Comment by maugre** [comment]
@ahmed-elsayed2017 modify your /etc/makepkg.conf file to have your system use all your CPU during build by adding a -j<number of cpu threads> in MAKEFLAGS. For example on my system:
```
#-- Make Flags: change this for DistCC/SMP systems
MAKEFLAGS="-j12"
```

**Comment by doitsujin** [comment]
Not sure what the hell the Manjaro package is doing then, my local build completes in 1:41 *minutes* on my Ryzen 2700X.

But yeah, if you don't want to mess around with patches, better wait for them to get merged into the master branch.

**Comment by ahmed-elsayed2017** [comment]
@maugre

What is the right value for my CPU?

I use Intel core I5 4200M

![screenshot_20181015_132626](https://user-images.githubusercontent.com/29783425/46948426-59738600-d07e-11e8-914d-d8bfba6b68e8.png)

**Comment by C0rn3j** [comment]
@ahmed-elsayed2017 

![image](https://user-images.githubusercontent.com/1641362/46948591-e4ed1700-d07e-11e8-8fce-935ffa6870b8.png)

**Comment by shmerl** [comment]
>  I think some of them are already in (updated Vulkan headers etc.), which makes applying them manually even harder.

Just apply patches 2-15 for radv for now, skip the first one.

**Comment by ahmed-elsayed2017** [comment]
@C0rn3j 
It doesn't give accurate answer but I will try j8 and see how much CPU usage will be during compiling.

**Comment by Bfgeshka** [comment]
@ahmed-elsayed2017 By the way, you don't want to enable too many concurrent make jobs if you do not have enough memory, keep that in mind. It may hit swap hard, especially on llvm compilation.

**Comment by HanFox** [comment]
I don't know how helpful I'm really being doing this (as if you can't edit the file yourself, you probably shouldn't trust an idiot on the internet giving you a random file with the ability to break your PC...), but for Arch Linux users using RADV here's the PKGBUILD I used: [PKGBUILD](https://gist.github.com/HanFox/86f1748fd38b4d6b0509ba99deac4ae9)

You need to download it and save it to a folder, you also need the patch files 2-15 from [here](https://patchwork.freedesktop.org/series/50968/) saved in the same folder and you need the [LICENSE](https://aur.archlinux.org/cgit/aur.git/plain/LICENSE?h=mesa-git) file from the AUR also saved in that folder.

Open a terminal/console in the folder and run `makepkg -i`

This will either work for you, or it won't. I shan't help you debug it if it doesn't, or if your PC explodes. You've been warned...

**Comment by SveSop** [comment]
@ahmed-elsayed2017 [https://ark.intel.com/products/76348/Intel-Core-i5-4200M-Processor-3M-Cache-up-to-3-10-GHz-](https://ark.intel.com/products/76348/Intel-Core-i5-4200M-Processor-3M-Cache-up-to-3-10-GHz-)
```
# of Cores 2
# of Threads 4 
```
Looks like its a dual-core w/hyperthreading. So, you should be using -j4 or -j5 (if plenty of ram, go for -j5).

**Comment by SveSop** [comment]
@doitsujin I also see Józef Kucia has added a patch to wine:
[https://source.winehq.org/patches/data/152604](https://source.winehq.org/patches/data/152604)
winevulkan: Update vk.xml to 1.1.88.

This seems to implement the winevulkan-transform-feedback.patch (+ more) i think.

**Comment by doitsujin** [comment]
I replaced the custom patch in the original post with a link to the official wine patch.

**Comment by ahmed-elsayed2017** [comment]
> @ahmed-elsayed2017 https://ark.intel.com/products/76348/Intel-Core-i5-4200M-Processor-3M-Cache-up-to-3-10-GHz-
> 
> ```
> # of Cores 2
> # of Threads 4 
> ```
> Looks like its a dual-core w/hyperthreading. So, you should be using -j4 or -j5 (if plenty of ram, go for -j5).

I used -j8 to compile lib32-mesa-git and it took 4.5 hours (half the time was before). Cpu usage didn't exceed 30%. I will try -j12 next time to see if it will increase cpu usage and being stable at the same time.

**Comment by SveSop** [comment]
@ahmed-elsayed2017 the "-j" option sets how many threads you use. It is not really beneficial to use "too many", or else one would just pop "-j100" and compile very fast...

You can easily try different settings, but afaik you will only loose time compiling with too many threads. Discussion about having "thread +1" vs "threads" - ie. for YOU "-j4" vs "-j5" is up to debate, and kind of non-conclusive :) 

To check how many cores your Linux install have found, you can do:
`lscpu`

The output for ME with a i7-8700K 6cores+6threads is:
```
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              12
On-line CPU(s) list: 0-11
Thread(s) per core:  2
Core(s) per socket:  6
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               158
Model name:          Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz
```
Here you see i have 12 cpu's, listed as 2 threads per core, and 6 cores per socket.. with 1 socket, and that is correct based on my model.

I use "-j12" when i compile... but have not benchmarked if i could earn a millisec or two from doing "-j13" :)

**Comment by shmerl** [comment]
Wine has [landed vulkan patches with VK_EXT_transform_feedback upstream](https://source.winehq.org/git/wine.git/commit/648bd17b81831e16e4d1466075353ae2637bf840), so no need to patch it manually anymore, just build Wine master. It should be available in Wine 3.19 release.

**Comment by Sukid** [comment]
Does this work with winelib versions or no? It might be unrelated, but when I use a winelib build(which at this point is 6 commits ahead of the release), the victory screens in Overwatch are blacked out. Works fine using the normal release. Admittedly this could be a result of one of the 6 commits to master since the .90 release but I figured I would mention this.

**Comment by doitsujin** [comment]
If there's a regression, please bisect it. I haven't heard any complaints about Overwatch since 0.90 though.

**Comment by SveSop** [comment]
@Sukid I cant get World of Warcraft to launch at all with a winelib build. Tested building winelib with dxvk-0.81, dxvk-0.90 and 629587bedd92ef645afc7818445960530179599d

Possibly totally unrelated, but given they both are Blizzard games...

**Comment by Guy1524** [comment]
@SveSop this thread is for discussion of stream output support in DXVK

**Comment by llyzs** [comment]
The Vulkan-Headers and Vulkan-Loader repos are not yet updated to 1.1.88. How do you compile dxvk-0.90 with transform feedback support?

**Comment by K0bin** [comment]
You don't have to compile DXVK yourself, just use release build 0.90. You'll need to compile Wine though.

**Comment by shmerl** [comment]
Compiling dxvk yourself doesn't need vulkan dev package at 1.1.88.

**Comment by minionprocyk** [comment]
The only thing you need to compile yourself will be your graphics drivers. Try pulling down the mesa3d repo, apply the patches provided by the OP, build the drivers, then you can verify the header is there with vulkaninfo | grep VK_EXT_t and you should see the transform feedback extension. Unless you link your new drivers to /usr/lib youll have to prefix vulkaninfo with the LD_LIBRARY_PATH, LIBGL_DRIVERS_PATH, and EGL_DRIVERS_PATH environment variables. 

OR 

you can just wait a while for the patches to be merged and distributed

**Comment by shmerl** [comment]
> Unless you link your new drivers to /usr/lib youll have to prefix vulkaninfo with the LD_LIBRARY_PATH, LIBGL_DRIVERS_PATH, and EGL_DRIVERS_PATH environment variables.

You can use Vulkan ICD file for that. I.e. something like:

**Comment by minionprocyk** [comment]
Oh. That's way more badass. Seems like the default for ubuntu 18 is /usr/local/share/vulkan/icd.d

The more you know.

**Comment by mati865** [comment]
@minionprocyk `/usr/local/share/vulkan/icd.d` doesn't even exist on Ubuntu by default.
`/usr/local` is for stuff installed via `make install`, it's common thing for all build systems to put it there by default. Being Ubuntu or not has nothing to do with it.

**Comment by minionprocyk** [comment]
@mati865 Sorry I thought it was obvious this was from a build since we're talking about compiling graphics drivers. 

> Being Ubuntu or not has nothing to do with it.
 good to know. I'm not very savy with differences across distros so I generally state the distro I work with.

**Comment by ahmed-elsayed2017** [comment]
Mesa 18.3 Devel with transform_feedback patch for Manjaro and Arch users

mesa
https://aur.archlinux.org/packages/mesa-transform-feedback-git

lib32-mesa
https://aur.archlinux.org/packages/lib32-mesa-transform-feedback-git/

**Comment by ahmed-elsayed2017** [comment]
https://github.com/doitsujin/dxvk/issues/719

**Comment by ahmed-elsayed2017** [comment]
@doitsujin Do you know if Mesa is going to release VK_EXT_transform_feedback patches in the next stable release 18.2.4?

**Comment by shmerl** [comment]
Not until tests are complete and passing.

https://patchwork.freedesktop.org/series/50969/

> This series adds support for the new VK_EXT_transform_feedback extension.
As far as I know, everything works but it's still all a bit experimental as
we don't have very good tests yet.  CTS tests are in-progress and we hope
to have decent testing soon.  We likely won't be landing them in master
until we get better testing as the testing is currently really sketchy.

I hope they'll do it in time before 18.3 though.

**Comment by doitsujin** [comment]
18.2.x releases are no feature updates anyway, so no.

**Comment by ahmed-elsayed2017** [comment]
> Not until tests are complete and passing.
> 
> https://patchwork.freedesktop.org/series/50969/
> 
> > This series adds support for the new VK_EXT_transform_feedback extension.
> > As far as I know, everything works but it's still all a bit experimental as
> > we don't have very good tests yet.  CTS tests are in-progress and we hope
> > to have decent testing soon.  We likely won't be landing them in master
> > until we get better testing as the testing is currently really sketchy.
> 
> I hope they'll merge it in time before 18.3 though.

I hope that too, because it doesn't work well right now on my AMD card

![screenshot_20181021_224447](https://user-images.githubusercontent.com/29783425/47434392-be2b9080-d7a2-11e8-94dd-d23ff47f044d.png)

![screenshot_20181023_010012](https://user-images.githubusercontent.com/29783425/47434426-d1d6f700-d7a2-11e8-870a-f7a20ff92e3c.png)

**Comment by ahmed-elsayed2017** [comment]
In the good side I tried Mafia III on Windows then I tried it on Manjaro with Mesa 18.3 git with RADV Patches for VK_EXT_transform_feedback, and the performance is better than Windows. The only downside right now is the graphic glitches, so I hope that it will be fix it soon.

**Comment by doitsujin** [comment]
**Update**: RADV just landed Transform Feedback support in mesa-git. No more manual patching required!

**Comment by shmerl** [comment]
Great! So it will be available in 18.3.

**Comment by C0rn3j** [comment]
Just got around to installing the 396.54.09 driver.

This patch also fixes non-rendering characters in Furi, awesome!

**Comment by 7oxicshadow** [comment]
I play quake champions and have just discovered that not only am I using the wrong NVIDIA driver version (410.73) but I am also using the wrong Vulkan driver (1.1.82) version but since moving to Wine 3.19 staging Quake champions now works perfectly without any crashes with Sorlag or any maps that were known to cause problems.

I can only assume that the game needs to be told that the extension is available without actually using it? Either way it will keep me playing until the NVIDIA drivers are updated officially :D

**Comment by SveSop** [comment]
@7oxicshadow Not entirely sure how nVidia does its versioning, but i guess 410.73 have vulkan 1.1.82, while the 396.54.09 has 1.1.85... However, from the LunarG site, the needed `VK_EXT_transform_feedback` for stream output came with 1.1.88? 
You can check the apiversion with: `vulkaninfo | grep -i apiversion`, and if you have the extension: `vulkaninfo | grep -i vulkaninfo | grep -i apiversion`

With 396.54.09 it reads:
```
apiVersion     = 0x401055  (1.1.85)
VK_EXT_transform_feedback           : extension revision  1
```
Guess the extension is kinda "retro-fitted" to work with the latest vulkan beta-driver from nVidia, since it in theory should be apiVersion 1.1.88 ?

**Comment by doitsujin** [comment]
Just ignore the Vulkan version, it doesn't mean much. The extension had been in development for several months anyway.

**Comment by enrique556** [comment]
@SveSop where you had:

> You can check the apiversion with: vulkaninfo | grep -i apiversion, and if you have the extension: vulkaninfo | grep -i vulkaninfo | grep -i apiversion

Did you mean:

`vulkaninfo | grep -i apiversion && vulkaninfo | grep -i VK_EXT_transform_feedback`

**Comment by SveSop** [comment]
Yeah, double grep was a typo :) Thx for clearing that up @enrique556

**Comment by dcraig327** [comment]
FYI. Found this on nVidia forums, hope it's useful:
[https://devtalk.nvidia.com/default/topic/1044231/linux/which-driver-to-use-long-short/post/5297592/#5297592](https://devtalk.nvidia.com/default/topic/1044231/linux/which-driver-to-use-long-short/post/5297592/#5297592)

**Comment by SveSop** [comment]
Interesting note from the changelog for 410.x driver:

> Removed support for the VK_NV_glsl_shader Vulkan extension.
>       This extension was added before the Vulkan ecosystem had mature
>       SPIR-V compilers and tools.  Now that the SPIR-V compilers and
>       tools have advanced, Vulkan developers are encouraged to use SPIR-V
>       instead of GLSL.
> 

Perhaps something to plan for @doitsujin ? I see references to this in various places...

**Comment by doitsujin** [comment]
> I see references to this in various places..

Where? I never used that extension.

**Comment by SveSop** [comment]
Possibly not USED for anything, so it could probably just stay quietly in /include/vulkan/vulkan_core.h
```
#define VK_NV_glsl_shader 1
#define VK_NV_GLSL_SHADER_SPEC_VERSION    1
#define VK_NV_GLSL_SHADER_EXTENSION_NAME  "VK_NV_glsl_shader"
```

**Comment by doitsujin** [comment]
Yeah, those will disappear when I update the headers the next time.

**Comment by florensie** [comment]
How do I get the right nvidia driver version on arch? I can't find it in the repos.

**Comment by C0rn3j** [comment]
415.18.02 is the version you should use nowadays.
aur/nvidia-vulkan or aur/nvidia-vulkan-dkms (you need linux-headers for this one assuming you use vanilla kernel)  is what you want.

**Comment by doitsujin** [comment]
415.18.02 is known to break all Unity Engine games on Wine, so if you have any of those, it's better to stay away from it. I'm not going to recommend it until this issue is sorted out.

**Comment by ahmed-elsayed2017** [comment]
I hope they don't make any new games with Unity Engine because all Unity games that I tried so far have problems with controllers and I have to use steam to config controllers to act as mouse and keyboard.

**Comment by C0rn3j** [comment]
If you mean via WINE then you may have some luck with https://github.com/kozec/dumbxinputemu and sc-controller. A bit OT though.

**Comment by doitsujin** [comment]
@C0rn3j Thanks, I'm aware of it and will test it later, will update the wiki if it runs well.

**Comment by doitsujin** [comment]
Status update:
- RADV and Nvidia support this extension in their mainline drivers.
- Intel doesn't yet, but patches are available.
- AMDVLK will probably support this quite soon as well.

**Comment by florensie** [comment]
@doitsujin Your post says 418.22 but I believe it's supposed to be 415.22.

**Comment by doitsujin** [comment]
Right, thanks.
